controller/tests/array_action/ds8k/test_array_mediator_ds8k.py

```diff
@@ -514,79 +536,70 @@ def test_create_snapshot_not_valid(self):
+        self._prepare_mocks_for_delete_snapshot()
         self.array.delete_snapshot("test_id")
-        flashcopy_id = volume_with_flashcopy_rel.flashcopy[0].id
-        self.client_mock.delete_flashcopy.assert_called_once_with(flashcopy_id)
+        self.client_mock.delete_volume.assert_called_once()
+        self.client_mock.delete_flashcopy.assert_called_once_with(self.flashcopy_response.id)
 
-    def test_delete_flashcopy_failed_with_ClientException(self):
+    def test_delete_snapshot_flashcopy_failed_with_ClientException(self):
+        self._prepare_mocks_for_delete_snapshot()
         self.client_mock.delete_flashcopy.side_effect = ClientException("500")
-        volume_with_flashcopy_rel = self._get_volume_with_flashcopy_relationship()
-        self.client_mock.get_volume.return_value = volume_with_flashcopy_rel
+        self.client_mock.get_volume.return_value = self.snapshot_response
         with self.assertRaises(ClientException):
             self.array.delete_snapshot("fake_name")
 
     def test_delete_snapshot_failed_with_ClientException(self):
+        self._prepare_mocks_for_delete_snapshot()
         self.client_mock.delete_volume.side_effect = ClientException("500")
         with self.assertRaises(array_errors.VolumeDeletionError):
             self.array.delete_snapshot("fake_name")
 
     def test_delete_snapshot_failed_with_NotFound(self):
+        self._prepare_mocks_for_delete_snapshot()
         self.client_mock.get_volume.side_effect = NotFound("404")
         with self.assertRaises(array_errors.ObjectNotFoundError):
             self.array.delete_snapshot("fake_name")
 
     def _prepare_mocks_for_copy_to_existing_volume(self):
-        volume = Munch(
-            {"cap": "1073741824",
-             "id": "0001",
-             "name": "test_name",
-             "pool": "fake_pool",
-             "tp": "ese",
-             "flashcopy": ""
-             }
-        )
+        volume = self.volume_response
         self.client_mock.get_volumes_by_pool.side_effect = [[volume], [Munch(
             {"cap": "1073741824",
              "id": "0002",
-             "name": "snapshot_name",
+             "name": "source_name",
              "pool": "fake_pool",
              "tp": "ese",
              "flashcopy": []
              }
         )]]
         self.client_mock.get_volume.return_value = volume
         self.client_mock.get_flashcopies_by_volume.side_effect = \
-            [[], [self.flashcopy_from_volume_response], [self.flashcopy_from_volume_response]]
-        self.client_mock.get_flashcopies.return_value = self.flashcopy_from_volume_response
+            [[], [self.flashcopy_response], [self.flashcopy_response]]
+        self.client_mock.get_flashcopies.return_value = self.flashcopy_response
         self.client_mock.create_flashcopy.return_value = self.flashcopy_response
         return volume
 
-    def test_extend_volume(self):
+    def test_copy_to_existing_volume_with_extend_volume(self):
```

Comment:
either split it to `test_copy_to_existing_volume_calls_extend_volume` and `test_copy_to_existing_volume_calls_create_flashcopy` , or choose a more general name, e.g. `test_copy_to_existing_volume_success`

Reply:
done